python/cuml/metrics/pairwise_distances.pyx

```diff
@@ -0,0 +1,273 @@
+    elif metric_str == 'euclidean':
+        return DistanceType.EucUnexpandedL2Sqrt
+    elif metric_str == 'haversine':
+        raise ValueError(" The metric: '{}', is not supported at this time."
+                         .format(metric_str))
+    elif metric_str == 'l2':
+        return DistanceType.EucUnexpandedL2Sqrt
+    elif metric_str == 'l1':
+        return DistanceType.EucUnexpandedL1
+    elif metric_str == 'manhattan':
+        return DistanceType.EucUnexpandedL1
+    elif metric_str == 'nan_euclidean':
+        raise ValueError(" The metric: '{}', is not supported at this time."
+                         .format(metric_str))
+    elif metric_str == 'sqeuclidean':
+        return DistanceType.EucUnexpandedL2
+    else:
+        raise ValueError("Unknown metric: {}".format(metric_str))
+
+
+@with_cupy_rmm
+def pairwise_distances(X, Y=None, metric="euclidean", handle=None,
+                       convert_dtype=True, **kwds):
+    """
+    Compute the distance matrix from a vector array `X` and optional `Y`.
+
+    This method takes either one or two vector arrays, and returns a distance
+    matrix.
+
+    If `Y` is given (default is `None`), then the returned matrix is the
+    pairwise distance between the arrays from both `X` and `Y`.
+
+    Valid values for metric are:
+
+    - From scikit-learn: ['cityblock', 'cosine', 'euclidean', 'l1', 'l2', \
+        'manhattan'].
+        Sparse matrices are not supported.
+    - From scipy.spatial.distance: ['sqeuclidean']
+        See the documentation for scipy.spatial.distance for details on this
+        metric. Sparse matrices are not supported.
+
+    Parameters
+    ----------
+    X : array-like (device or host) of shape (n_samples_x, n_features)
+        Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+        ndarray, cuda array interface compliant array like CuPy
+
+    Y : array-like (device or host) of shape (n_samples_y, n_features),\
+        optional
+        Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+        ndarray, cuda array interface compliant array like CuPy
+
+    metric : {"cityblock", "cosine", "euclidean", "l1", "l2", "manhattan", \
+        "sqeuclidean"}
+        The metric to use when calculating distance between instances in a
+        feature array.
+
+    convert_dtype : bool, optional (default = True)
+        When set to True, the method will, when necessary, convert
+        Y to be the same data type as X if they differ. This
+        will increase memory used for the method.
+
+    Returns
+    -------
+    D : array [n_samples_x, n_samples_x] or [n_samples_x, n_samples_y]
+        A distance matrix D such that D_{i, j} is the distance between the
+        ith and jth vectors of the given matrix `X`, if `Y` is None.
+        If `Y` is not `None`, then D_{i, j} is the distance between the ith
+        array from `X` and the jth array from `Y`.
+
+    Examples
+    ---------
+        >>> import numpy as np
```

Comment:
Per the very brief discussion on the team slack channel, we should use device arrays in our examples, rather than numpy. This is another update that we will eventually need to scrape through and make consistent. I’d personally recommend cupy rather than numpy but cudf would work as well.

Reply:
Thanks for the feedback. I've updated the example